source/MaterialXRender/Handlers/ImageHandler.h

```diff
@@ -46,6 +60,18 @@ class ImageDesc
     }
 };
 
+/// Structure containing harware image description restrictions
+class HwImageDescRestrictions
+{
+  public:
+    /// List of channel counts that can be supported
+    std::set<unsigned int> supportedChannelCounts;
+    /// List of base types that can be supported
+    std::set<ImageDesc::BaseType> supportedBaseTypes;
+    /// List of image types that can be support
+    std::set<ImageDesc::ImageType> supportedImageTypes;
+};
```

Comment:
I'm not sure about this generalization to use a restrictions struct. Currently it's only used for one thing, the half->float conversion in OIIO loader, and even in that case there is implied knowledge about how the conversion is done. (See more comments below). I wonder if it would not be better to keep this internal to the OIIO loader as it was. 

With the generalization as it is presented now an API user would assume this is used and respected by all image loaders.